### PR TITLE
Improve messageCreate.js

### PR DESCRIPTION
**Pull request by XstreamSpeed**
**Please describe the changes this PR makes and why it should be merged:**

**Status and versioning classification:**
Node.js: v16.7.0 | discord.js: v13.8.0 | branch: v5

**Preview:**
Screenshot_9

- Code changes have been tested against the Discord API, or there are no code changes

### CODE REVIEW

**Review by DevAmiyo — APPROVED** [review]

File: events/messageCreate.js [hunk 1]
```diff
@@ -1,18 +1,30 @@
...
 module.exports = async (client, message) => {
-	const mention = new RegExp(`^<@!?${ client.user.id }>( |)$`);
+    const mention = new RegExp(`^<@!?${ client.user.id }>( |)$`);
+
+    const buttons = new MessageActionRow()
+	.addComponents(
+      new MessageButton()
+		.setStyle("LINK")
+		.setLabel('Invite me')
+		.setURL(`https://discord.com/oauth2/authorize?client_id=${client.config.clientId}&permissions=${client.config.permissions}&scope=bot%20applications.commands`),
```
Comment (DevAmiyo) [inline events/messageCreate.js:11]: Use scopes from config
```js
.setURL(`https://discord.com/oauth2/authorize?client_id=${client.config.clientId}&permissions=${client.config.permissions}&scope=${client.config.scopes.toString().replace(/,/g, '%20')}`),
```

**Review by DevAmiyo — CHANGES_REQUESTED** [review]

**Comment by DevAmiyo** [comment]
I have made some changes
> - Use scopes from config
> - Keep hyperlink in re-invite
> - Add dashboard button shown in screenshot as its a nice feature to add

```js
const { MessageEmbed, MessageButton, MessageActionRow } = require("discord.js");

module.exports = async (client, message) => {
	const mention = new RegExp(`^<@!?${ client.user.id }>( |)$`);
	const invite = `https://discord.com/oauth2/authorize?client_id=${client.config.clientId}&permissions=${client.config.permissions}&scope=${client.config.scopes.toString().replace(/,/g, '%20')}`;
	const buttons = new MessageActionRow()
	.addComponents(
		new MessageButton()
			.setStyle("LINK")
			.setLabel('Invite me')
			.setURL(invite),
		new MessageButton()
			.setStyle("LINK")
			.setLabel('Support server')
			.setURL(client.config.supportServer),
		new MessageButton()
			.setStyle("LINK")
			.setLabel('Dashboard')
			.setURL(client.config.website),
	);

	if (message.content.match(mention)) {
		const mentionEmbed = new MessageEmbed()
		.setColor(client.config.embedColor)
		.setDescription(
			`My prefix on this server is \`/\` (Slash Command).\nTo get started you can type \`/help\` to see all my commands.\nIf you can't see it, Please [re-invite](invite) me with the correct permissions.`
		);

		message.channel.send({
			embeds: [mentionEmbed],
			components: [buttons],
		});
	}
};
```

**Comment by XstreamSpeed** [comment]
> I have made some changes
> 
> > * Use scopes from config
> > * Keep hyperlink in re-invite
> > * Add dashboard button shown in screenshot as its a nice feature to add
> 
> ```js
> const { MessageEmbed, MessageButton, MessageActionRow } = require("discord.js");
> 
> module.exports = async (client, message) => {
> 	const mention = new RegExp(`^<@!?${ client.user.id }>( |)$`);
> 	const invite = `https://discord.com/oauth2/authorize?client_id=${client.config.clientId}&permissions=${client.config.permissions}&scope=${client.config.scopes.toString().replace(/,/g, '%20')}`;
> 	const buttons = new MessageActionRow()
> 	.addComponents(
> 		new MessageButton()
> 			.setStyle("LINK")
> 			.setLabel('Invite me')
> 			.setURL(invite),
> 		new MessageButton()
> 			.setStyle("LINK")
> 			.setLabel('Support server')
> 			.setURL(client.config.supportServer),
> 		new MessageButton()
> 			.setStyle("LINK")
> 			.setLabel('Dashboard')
> 			.setURL(client.config.website),
> 	);
> 
> 	if (message.content.match(mention)) {
> 		const mentionEmbed = new MessageEmbed()
> 		.setColor(client.config.embedColor)
> 		.setDescription(
> 			`My prefix on this server is \`/\` (Slash Command).\nTo get started you can type \`/help\` to see all my commands.\nIf you can't see it, Please [re-invite](invite) me with the correct permissions.`
> 		);
> 
> 		message.channel.send({
> 			embeds: [mentionEmbed],
> 			components: [buttons],
> 		});
> 	}
> };
> ```

I will use the scopes from the config, thank you for the suggestion, I don't know why you'd want a hyperlink instead of a nice button, we'll decide it later and for the dashboard, I kind of added a scope in my own bot for the dashboard if that seems fine I'll commit it in.

**Comment by DevAmiyo** [comment]
> I will use the scopes from the config, thank you for the suggestion, I don't know why you'd want a hyperlink instead of a nice button, we'll decide it later and for the dashboard, I kind of added a scope in my own bot for the dashboard if that seems fine I'll commit it in, some people don't know how to get the link to show the dashboard so I'm a bit skeptical on that.

- I'm not asking to remove the button but the button says "Invite Me" but the sentence is "Please reinvite me with the correct permissions.". So, I just think that the button can stay and the reinvite text could also have a hyperlink because its asking the user to reinvite the bot into the same server. Idk if it makes sense, but yeah.
- I agree with you, but eventually when the dashboard is fully built, we can add a button to view the dashboard or maybe add a condition if the value of website in config is not empty or localhost only then it would show the button. We can decide on that later on.
- Otherwise, everything looks good to me.

**Review by JotaroKujo0525 — APPROVED** [review]

**Comment by XstreamSpeed** [comment]
> > I will use the scopes from the config, thank you for the suggestion, I don't know why you'd want a hyperlink instead of a nice button, we'll decide it later and for the dashboard, I kind of added a scope in my own bot for the dashboard if that seems fine I'll commit it in, some people don't know how to get the link to show the dashboard so I'm a bit skeptical on that.
> 
> * I'm not asking to remove the button but the button says "Invite Me" but the sentence is "Please reinvite me with the correct permissions.". So, I just think that the button can stay and the reinvite text could also have a hyperlink because its asking the user to reinvite the bot into the same server. Idk if it makes sense, but yeah.
> * I agree with you, but eventually when the dashboard is fully built, we can add a button to view the dashboard or maybe add a condition if the value of website in config is not empty or localhost only then it would show the button. We can decide on that later on.
> * Otherwise, everything looks good to me.

Alright then!

**Comment by XstreamSpeed** [comment]
Bro why does this happen every. single. time, i make a pr but that pr is the only one that is not merged :/

**Comment by DevAmiyo** [comment]
> Bro why does this happen every. single. time, i make a pr but that pr is the only one that is not merged :/

lmao 😂

**Comment by XstreamSpeed** [comment]
> > Bro why does this happen every. single. time, i make a pr but that pr is the only one that is not merged :/
> 
> lmao 😂

watch this pr get closed

**Comment by Saud-97** [comment]
> Bro why does this happen every. single. time, i make a pr but that pr is the only one that is not merged :/

Its not actively maintained probably they are busy. I got PRs from 18 days ago lol.

**Comment by Saud-97** [comment]
Btw did you try and kick the bot and then clicked the invite me button? Does it work when bot is not in the server?

**Comment by DevAmiyo** [comment]
> Btw did you try and kick the bot and then clicked the invite me button? Does it work when bot is not in the server?

Yup, just checked. The button works regardless if the bot is in the server.

**Comment by XstreamSpeed** [comment]
> > > Bro why does this happen every. single. time, i make a pr but that pr is the only one that is not merged :/
> > 
> > 
> > Its not actively maintained probably they are busy. I got PRs from 18 days ago lol.
> 
> yes, and also cause it needs testing before merging to master branch. although, at this point I don't test @XstreamSpeed PR anymore. I rather let Sudhan or someone else do it. as their commit (no offense) is _usually_ untested and previously broke multiple stuff. also the code formatting is a bit off. _this is probably due to @XstreamSpeed using github instead of their IDE to commit these changes._
> 
> IMG_2283

This is tested as the image is shows above

**Comment by Neko-Life** [comment]
> I first write the code I IDE and then I publish it on github

By "publish" you mean "copy the codes from the IDE and paste it on my fork" am I right

**Comment by XstreamSpeed** [comment]
> > I first write the code I IDE and then I publish it on github
> 
> By "publish" you mean "copy the codes from the IDE and paste it on my fork" am I right

Kind of, but I reformat the code in github since github messes up tabs and spaces

**Comment by DevAmiyo** [comment]
> Kind of, but I reformat the code in github since github messes up tabs and spaces

Why don't you just use [Github Desktop](https://desktop.github.com) and push it directly to the fork?

**Comment by DarrenOfficial** [comment]
Github desktop is bloated tbh, it's better to just use [Git cli](https://git-scm.com/)

**Comment by XstreamSpeed** [comment]
> > Kind of, but I reformat the code in github since github messes up tabs and spaces
> 
> Why don't you just use [Github Desktop](https://desktop.github.com) and push it directly to the fork?

tbh I would use vscode and commit changes from there but i guess im too lazy

**Comment by XstreamSpeed** [comment]
waiting for the day where darren or sudhan do something with this..

**Comment by JotaroKujo0525** [comment]
so er, do we merge this or nah

**Comment by XstreamSpeed** [comment]
> so er, do we merge this or nah

i mean its tested and shit..

**Review by SudhanPlayz — APPROVED** [review]

**Pull request merged**